operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Context.java

```diff
@@ -43,4 +44,12 @@ <R> Optional<R> getSecondaryResource(Class<R> expectedType,
    * ExecutorService initialized by framework for workflows. Used for workflow standalone mode.
    */
   ExecutorService getWorkflowExecutorService();
+
+  /**
+   * Retrieves the primary resource cache.
+   *
+   * @return the {@link IndexerResourceCache} associated with the associated {@link Reconciler} for
+   *         this context
+   */
+  IndexerResourceCache<P> getPrimaryCache();
```

Comment:
I don't like the fact that the `IndexerResourceCache` interface allows access to creating new indices at any time when these would actually only take effect whenever the associated informers are initialized so I think we should split this interface into a read-only part and one to manage indexers. The alternative would be to only expose `ResourceCache` instead of `IndexerResourceCache`.

Reply:
I've added an `IndexedResourceCache` interface that extends `ResourceCache` to add the indexed-based retrieval and moved both classes to the reconciler package since they'd now be part of the API… It shouldn't break anything since the only thing that was used externally was the `IndexerResourceCache` interface.